client/resource_group/controller/model.go

```diff
@@ -133,9 +138,11 @@ func newSQLCalculator(cfg *Config) *SQLCalculator {
 	return &SQLCalculator{Config: cfg}
 }
 
-// Trickle ...
-// TODO: calculate the SQL CPU cost and related resource consumption.
+// Trickle Update Sql Layer CPU consumption.
 func (dsc *SQLCalculator) Trickle(ctx context.Context, consumption *rmpb.Consumption) {
+	delta := getSQLProcessCPUTime() - consumption.SqlLayerCpuTimeMs
```

Comment:
The `consumption.SqlLayerCpuTimeMs` seems should be zero here, why use `getSQLProcessCPUTime()` to minus it?

Reply:
Since the `get pid` function of `gosigar` can only get the cpu time from the start of the thread to this point.
And to satisfy the purpose of getRUValue/getRawValue to remove tokens from delta
so the last cpu time is passed before the trickle.

> And to satisfy the purpose of getRUValue/getRawValue to remove tokens from delta

Because before this pr the tidb cpu is also to count RU's, but Actually, I'm not quite sure what the purpose of this now.
Do you think i need to refine the `updateRunState` and `trickle` code?